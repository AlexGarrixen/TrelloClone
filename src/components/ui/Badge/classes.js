const classes = {
  default: 'badge',
  // class applied if color=green
  colorGreen: 'badge--green',
  // class applied if color=yellow
  colorYellow: 'badge--yellow',
  // class applied if color=orange
  colorOrange: 'badge--orange',
  // class applied if color=ref
  colorPurple: 'badge--purple',
  // class applied if color=gray
  colorBlue: 'badge--blue',
};

export default classes;
