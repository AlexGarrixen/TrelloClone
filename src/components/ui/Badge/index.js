import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import capitalize from '../../../utils/capitalize';
import classes from './classes';

const Badge = ({
  children,
  className,
  component: Component = 'span',
  color = 'blue',
  ...other
}) => (
  <Component
    className={clsx(
      classes.default,
      [`${classes[`color${capitalize(color)}`]}`],
      className
    )}
    {...other}
  >
    {children}
  </Component>
);

Badge.propTypes = {
  children: PropTypes.node,
  component: PropTypes.elementType,
  color: PropTypes.string,
};

export default Badge;
