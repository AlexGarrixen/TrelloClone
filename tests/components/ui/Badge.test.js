import React from 'react';
import { mount } from 'enzyme';
import Badge from '../../../src/components/ui/Badge';

describe('<Badge />', () => {
  test('Render Success', () => {
    const wrapper = mount(<Badge>badge</Badge>);

    expect(wrapper).toMatchSnapshot();
  });

  test('Test with props default', () => {
    const wrapper = mount(<Badge>badge</Badge>);
    const element = wrapper.find('.badge');

    expect(element.hasClass('badge badge--blue')).toBe(true);
    expect(element.type()).toBe('span');
    expect(element.text()).toBe('badge');
  });

  test('Test with prop component', () => {
    const wrapper = mount(<Badge component='article'>badge</Badge>);
    const element = wrapper.find('.badge');

    expect(element.type()).toBe('article');
  });

  test('Test with prop color', () => {
    const wrapper = mount(<Badge color='purple'>badge</Badge>);
    const element = wrapper.find('.badge');

    expect(element.hasClass('badge--purple')).toBe(true);
  });
});
