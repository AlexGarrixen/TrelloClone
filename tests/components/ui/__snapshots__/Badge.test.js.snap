// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> Render Success 1`] = `
<span
  className="badge badge--blue"
>
  badge
</span>
`;
